# Example 3

Figure Q2 illustrates a portion of a cover plate bolled to the end of a thick-walled cylindrical pressure vessel. A total of Nb bolts are to be used to resist a separating force P= 200 kN.
Determine;

i) the joint constant, C

ii) the number of bolt, Nb for the connection

iii) the force on the bolt, Fb and does the design safe or fail

iv) the suitable metric specification class number if M30 x 2 mm bolt is used to resist the
same separating force using the same amount of bolt in (ii).

Design Assumptions:. The connection is reusable. A bolt safety factor of n = 1.5 is used.
The modulus of elasticity of the parts Is assumed to be half of the modulus of elasticity of
the steel bolt.

In [28]:
# Required
from math import pi, e, pow, log

In [29]:
# All units in m
print('Coarse thread Class 8.8, Sp=600MPa')
Sp = 600E+6
d = 0.02
p = 2.5
flange_diameter=0.25
Eb = 200E+9
PLoad = 200E+3
print('Reusable connection, F=0.75Fp')
print('Ep=1/2Eb')
# From diagram
L = 0.05
FOS=1.5

Coarse thread Class 8.8, Sp=600MPa
Reusable connection, F=0.75Fp
Ep=1/2Eb


i) the joint constant, C

$$C=\frac{k_b}{k_b+k_p}$$

In [30]:
print('Stiffness constant for bolts, kb (Eb is left unknown)')

Ab=pi*pow(d,2)/4
# kb=Ab*Eb/L
kb=Ab/L
print('kb=', f"{kb:.3e}Eb")

Stiffness constant for bolts, kb (Eb is left unknown)
kb= 6.283e-03Eb


In [31]:
print('Stiffness constant for parts, kp (Ep is left unknown)')

kp=(0.58*pi*d)/(2*log(5*(((0.58*L)+(0.5*d))/((0.58*L)+(2.5*d)))))

print('kp=', f"{kp:.3e}Ep")

Stiffness constant for parts, kp (Ep is left unknown)
kp= 2.017e-02Ep


$$C=\frac{6.283\times10^{-3}E_b}{6.283\times10^{-3}E_b+0.02E_p}$$

when $E_p=\frac{1}{2}E_b$

$$C=\frac{6.283\times10^{-3}E_b}{6.283\times10^{-3}E_b+0.02(0.5\times E_b)}$$

In [32]:
new_kp = kp/2
C = kb/(kb+new_kp)
print(f'C= {C}')

C= 0.3839098600451956


ii) the number of bolt, Nb for the connection

If more than one bolt
$$F_b=\frac{CP}{Nb}+F_i$$

In [33]:
print('From Table 2.1, d=20mm, At=245mm^2')
print('From Table 2.2, Class 8.8, Sp=600MPa')
At=245E-6
Fp=Sp*At
Fi = 0.75*Fp #Reusable joint
print(f'Fi= {Fi/1000}kN')

From Table 2.1, d=20mm, At=245mm^2
From Table 2.2, Class 8.8, Sp=600MPa
Fi= 110.25kN


In [34]:
print('To find number of bolt, need to assume Total Force on bolt = Proof load')

To find number of bolt, need to assume Total Force on bolt = Proof load


$$S_pA_t=\frac{CPn}{Nb}+F_i$$

$$S_pA_t=\frac{(0.384)(200\times10^3)}{Nb}+110.25\times10^3$$
$$(600\times10^6)(245\times10^{-6})=\frac{(0.384)(200\times10^3)}{Nb}+110.25\times10^3$$

In [35]:
Nb=C*PLoad*FOS/((Sp*At)-Fi)
print(f'Nb={Nb}')
print(f'Therefore, No. of bolts = 4 bolts')
Nb=4

Nb=3.13395804118527
Therefore, No. of bolts = 4 bolts


iii) the force on the bolt, Fb and does the design safe or fail

$$S_pA_t=\frac{CPn}{Nb}+F_i$$

In [ ]:
Fb=(C*PLoad*FOS/Nb)+Fi
print(f'Total force on bolt, Fb={Fb/1000}kN')

Total force on bolt, Fb=139.04323950338966kN


Check $F_b > P$

Since $F_b < P$ the design will fail.

iv) the suitable metric specification class number if M30 x 2 mm bolt is used to resist the
same separating force using the same amount of bolt in (ii).

In [37]:
print('Metric specification means new class of bolt. Currently Class 8.8')

Metric specification means new class of bolt. Currently Class 8.8


Use
$$S_pA_t=\frac{CPn}{Nb}+F_i$$
$$S_pA_t=\frac{CPn}{Nb}+0.75S_pA_t$$

In [38]:
print('New diameter is used. kb will changed. Joint stiffness factor will also changed')
d = 0.03
Ab=pi*pow(d,2)/4
# kb=Ab*Eb/L
kb=Ab/L
print('kb=', f"{kb:.3e}Eb")

New diameter is used. kb will changed. Joint stiffness factor will also changed
kb= 1.414e-02Eb


In [39]:
new_kp = kp/2
C = kb/(kb+new_kp)
print(f'C= {C}')

C= 0.5836911841032998


$$S_pA_t=\frac{CPn}{Nb}+0.75S_pA_t$$
Where $A_t = 621mm^2$
$$0.25S_p(621\times10^{-6})=\frac{(0.584)(200\times10^3)\times(1.5)}{4}$$

In [43]:
At=621E-6
Sp=(C*PLoad*FOS/Nb)/(0.25*At)
print(f'Sp={Sp/1E6}MPa')
print('Class number 4.8')

Sp=281.9764174412077MPa
Class number 4.8
